src/vanilla/Model/CodeModelTS.cs

```diff
@@ -391,6 +394,110 @@ public string RequiredConstructorParametersTS
             }
         }
 
+        public virtual string GenerateConstructor(string superParameterList, Action<TSBlock> guardChecks = null, Action<TSBlock> implementation = null)
```

Comment:
Please introduce the difference between a parameter and an argument. It will make things much clearer.
```suggestion
        public virtual string GenerateConstructor(string superArgumentList, Action<TSBlock> guardChecks = null, Action<TSBlock> implementation = null)
```